### PR TITLE
Allow mounting journal paths on DMs

### LINKED ISSUE #208: Mounting paths to DMs
As a moderator who likes to be notified of more things than most people do, I'd like to be able to mount logger paths to my own DMs. This would mean I'd be able to get instant notifications about things only I care about, without disturbing or notifying others.

### PR DESCRIPTION
**Pull request by ammongit**
Fixes #208 
~~Requires #236 to be merged first~~

### CODE REVIEW

**Comment by ammongit** [comment]
journal PMs

File: futaba/journal/impl/direct_message.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+A Listener that DMs messages to the configured Discord user.
+"""
+
+import logging
+
+from futaba.permissions import mod_perm
```
Comment (Ma-wa-re) [inline futaba/journal/impl/direct_message.py]: This isn't used anywhere in the file. So should be removed.

File: futaba/journal/impl/direct_message.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        Send the message to the given channel, applying the icon if applicable.
+        """
+
+        if guild is not None:
+            content = f"**[{guild.name}]** {content}"
+
+            # Don't send journal events if they're not a mod
+            member = guild.get_member(self.user.id)
+            if member is None or not member.guild_permissions.manage_channels:
```
Comment (Ma-wa-re) [inline futaba/journal/impl/direct_message.py]: Is this where `mod_perm` was to be used?
If so it should be as if we want to change the permission needed for mod we wouldn't need to change this line

Comment (ammongit) [reply futaba/journal/impl/direct_message.py]: Correct: `*_perm` takes a `Context`, not a `Member`. There wasn't any reasonable way to produce one in this circumstance.

Do you have a suggestion of a more portable way to detect if a member is a moderator?

**Pull request merged**